src/components/DockFilter.js

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+
+const DockFilter = ({ onDockSelect, dockNumbers }) => {
+  const [dockNumber, setDockNumber] = useState('');
+
+  const handleSelectionChange = (selectedDockNumber) => {
+    onDockSelect(selectedDockNumber);
+  };
+
+  const handleChange = (e) => {
+    const newDockNumber = e.target.value;
+    setDockNumber(newDockNumber);
+    handleSelectionChange(newDockNumber);
+  };
+
+  return (
+    <form className="dock-select-form" onSubmit={(e) => e.preventDefault()}>
+      <label htmlFor="dock-select" className="dock-select-label">
```

Comment:
can you simply use `for`? https://github.com/itu-tid/lecture-notes/blob/main/Tech-TopUps/5-Coding-Practices.md#associate-labels-with-inputs-explicitly-with-for

Reply:
That's a great observation, it does work.